flexmeasures/cli/data_add.py

```diff
@@ -802,6 +808,141 @@ def create_forecasts(
+)
+@click.option(
+    "--until",
+    "end_str",
+    required=True,
+    help="Schedule ends at this datetime. Follow up with a timezone-aware datetime in ISO 6801 format.",
+)
+@click.option(
+    "--soc-at-start",
+    "soc_at_start",
+    type=float,
+    required=True,
+    help="State of charge (in MWh) at the start of the schedule.",
+)
+@click.option(
+    "--soc-target",
+    "soc_target_strings",
+    type=click.Tuple(types=[float, str]),
+    multiple=True,
+    required=False,
+    help="Target state of charge (in MWh) at some datetime. Follow up with a float value and a timezone-aware datetime in ISO 6081 format."
+    " This argument can be given multiple times."
+    " For example: --soc-target 32.8 2022-02-23T13:40:52+00:00",
+)
+@click.option(
+    "--soc-min",
+    "soc_min",
+    type=float,
+    required=False,
+    help="Minimum state of charge (in MWh) for the schedule.",
+)
+@click.option(
+    "--soc-max",
+    "soc_max",
+    type=float,
+    required=False,
+    help="Maximum state of charge (in MWh) for the schedule.",
+)
+@click.option(
+    "--roundtrip-efficiency",
+    "roundtrip_efficiency",
+    type=float,
+    required=False,
+    help="Round-trip efficiency (e.g. 0.85) to use for the schedule. Defaults to 1 (no losses).",
+)
+def create_schedule(
+    power_sensor_id: int,
+    factor_sensor_id: int,
+    start_str: str,
+    end_str: str,
+    soc_at_start: float,
+    soc_target_strings: List[Tuple[float, str]],
+    soc_min: Optional[float],
+    soc_max: Optional[float],
+    roundtrip_efficiency: Optional[float],
+):
+    """Create a new schedule for a given power sensor.
+
+    Current limitations:
+    - only supports battery assets and Charge Points
+    - only supports datetimes on the hour or a multiple of the sensor resolution thereafter
+    """
+
+    # Parse input
+    power_sensor: Sensor = Sensor.query.filter(
+        Sensor.id == power_sensor_id
+    ).one_or_none()
+    if power_sensor is None:
+        click.echo(f"No sensor found with Id {power_sensor_id}.")
+        raise click.Abort()
+    if not power_sensor.measures_power:
+        click.echo(f"Sensor with Id {power_sensor_id} is not a power sensor.")
+        raise click.Abort()
+    factor_sensor: Sensor = Sensor.query.filter(
+        Sensor.id == factor_sensor_id
+    ).one_or_none()
+    if factor_sensor is None:
+        click.echo(f"No sensor found with Id {factor_sensor_id}.")
+        raise click.Abort()
+    start = pd.Timestamp(start_str)
+    end = pd.Timestamp(end_str)
+    for attribute in ("min_soc_in_mwh", "max_soc_in_mwh"):
+        try:
+            check_required_attributes(power_sensor, [(attribute, float)])
+        except MissingAttributeException:
+            click.echo(f"{power_sensor} has no {attribute} attribute.")
+            raise click.Abort()
+    soc_targets = pd.Series(
+        np.nan,
+        index=pd.date_range(
```

Comment:
Two different issues at play here, but first some clarifications.
- ISO8601 handles timezone offsets (like `+01:00`), but does not deal with IANA timezone names (like `Europe/Amsterdam`).
- All FlexMeasures datetime input should follow the ISO8601 standard.
- FlexMeasures stores the relevant IANA timezone name as a sensor property.

A Pandas DatetimeIndex can start in one timezone offset and end in another, but only if matched to an IANA timezone name. Here's an example for our upcoming DST transition:

```
pd.date_range(pd.Timestamp("2022-03-27 00:00+01").tz_convert("Europe/Amsterdam"), pd.Timestamp("2022-03-27 04:00+02").tz_convert("Europe/Amsterdam"), freq="1H")
DatetimeIndex([
    '2022-03-27 00:00:00+01:00',
    '2022-03-27 01:00:00+01:00',
    '2022-03-27 03:00:00+02:00',
    '2022-03-27 04:00:00+02:00'
],
dtype='datetime64[ns, Europe/Amsterdam]', freq='H')
```
This would fail without the timezone conversions, which imo is just Pandas being cautious for practical reasons. https://github.com/FlexMeasures/flexmeasures/pull/372/commits/93519446603c01254cec4391ad136a28e3c135c0 fixes this.

The second issue (the nonsensical warnings about trimming the scheduling window) was due to a buggy if-statement that checks whether the available prices match up nicely with the query window. Fixed in https://github.com/FlexMeasures/flexmeasures/pull/372/commits/95d44e51e1b6bfa2e1a8bd06fdbdc93c8076a268.